### PR TITLE
Pylons - Fix old weapons not being removed

### LINKED ISSUE #6088: RHS Helicopters Pylons Configuration Ghost Pylons
**Arma 3 Version:** `1.80.143869' (stable)

**CBA Version:** `3.5` (stable)

**ACE3 Version:** `3.12.1` (stable)

**Mods:**
```
- CBA_A3
- ace
- RHSUSAF
- RHSAFRF
- RHSGREF
- RHSSAF
```

**Description:**
- When using Configure Pylons on RHS helicopters, any previous pylons that are removed are still in the weapon selection when cycling weapons. I.e. equip Hellfires, unequip Hellfires, Hellfires still appear in weapon selection but with no ammo.

**Steps to reproduce:**
- Load mods
- Place down RHS helicopter in editor (ex. AH-64D or AH-1Z) and a HEMTT Ammo Truck
- Configure pylons and change pylon or remove pylon armament
- Get into pilot or gunner seat and cycle weapons

**Where did the issue occur?**
- Dedicated, Self Hosted Multiplayer, and Editor

**Placed Modules:**
- None

### PR DESCRIPTION
**Pull request by PabstMirror**
Fix  #6088
- Use removeWeaponTurret instead of removeWeaponGlobal
- Fix "Too Far" warning showing on completion

Tag @654wak654

I have no idea why `removeWeaponGlobal` doesn't work for pilot seat of some RHS helis, but `removeWeaponTurret` works just fine.

### CODE REVIEW

File: addons/pylons/XEH_postInit.sqf [hunk 1]
```diff
@@ -28,8 +28,17 @@
...
     [QGVAR(setPylonLoadOutEvent), {
-        params ["_aircraft", "_pylonIndex", "_pylon", "_turret"];
+        params ["_aircraft", "_pylonIndex", "_pylon", "_turret", "_removeWeapon"];
+        TRACE_5("setPylonLoadOutEvent",_aircraft,_pylonIndex,_pylon,_turret,_removeWeapon);
         _aircraft setPylonLoadOut [_pylonIndex, _pylon, false, _turret];
+        if (_removeWeapon != "") then {
+            {
+                if (_aircraft turretLocal _x) then {
+                    TRACE_3("removing",_aircraft,_x,_removeWeapon);
+                    _aircraft removeWeaponTurret [_removeWeapon, [-1]];
```
Comment (Dystopian) [inline addons/pylons/XEH_postInit.sqf]: `[-1]` -> `_x` ? I didn't test yet but it looks strange.

File: addons/pylons/functions/fnc_showDialog.sqf [hunk 2]
```diff
@@ -149,7 +149,10 @@ if (!GVAR(isCurator)) then {
         isNull (GVAR(currentAircraft) getVariable [QGVAR(currentUser), objNull]) ||
         {(ace_player distanceSqr GVAR(currentAircraft)) > GVAR(searchDistanceSqr)}
     }, {
-        [localize LSTRING(TooFar), false, 5] call EFUNC(common,displayText);
+        TRACE_3("disconnect/far",GVAR(currentAircraft), ace_player distance GVAR(currentAircraft),GVAR(currentAircraft) getVariable QGVAR(currentUser));
```
Comment (654wak654) [inline addons/pylons/functions/fnc_showDialog.sqf]: There is a rogue space in the macro.

Comment (654wak654) [inline addons/pylons/functions/fnc_showDialog.sqf]: There is a rogue space in the macro.

File: addons/pylons/functions/fnc_configurePylons.sqf [hunk 3]
```diff
@@ -27,16 +27,19 @@ if (_currentPylon == count _pylonsToConfigure) exitWith {};
     {
         (_this select 0) params ["_pylonsToConfigure", "_currentPylon"];
         private _pylonIndex = _pylonsToConfigure select _currentPylon;
+        TRACE_2("",_currentPylon,_pylonIndex);
 
         // Remove the weapon of current pylon from aircraft IF weapon is only on this pylon
+        private _removeWeapon = "";
```
Comment (654wak654) [inline addons/pylons/functions/fnc_configurePylons.sqf]: Maybe call it `_weaponToRemove`?

**Pull request merged**